__docs__/wonder-blocks-theming/use-scoped-theme.mdx

```diff
@@ -0,0 +1,62 @@
+import {Meta, Canvas} from "@storybook/blocks";
+
+import {tokens, useScopedTheme} from "@khanacademy/wonder-blocks-theming";
+
+import {ViewWithTheme} from "./theme-examples";
+
+<Meta title="Theming / useScopedTheme" />
+
+# `useScopedTheme`
+
+`useScopedTheme` is a custom hook that allow us to get the theme based on a
+dynamic theme context.
+
+```ts
+function useScopedTheme<T>(themeContext: React.Context<T>): T;
+```
+
+## Usage
+
+```tsx
+import {useScopedTheme, tokens} from "@khanacademy/wonder-blocks-theming";
+
+// 👉 1. Define the theme.
+const defaultTheme = {
+  color: {
+    bg: {
+      primary: tokens.color.white,
+    },
+    text: {
+      primary: tokens.color.offBlack,
+    },
+  },
+  ...,
+};
+
+const themes = {
+  default: defaultTheme,
+};
+
+// 👉 2. Create the theme context scoped to this component.
+const ComponentThemeContext = React.createContext(themes.default);
```

Comment:
Are you planning to change this to use the custom `createThemeContext` that you wrote in #1977?

Reply:
ohh good point! I'll replace it here.